In [ ]:
import pandas as pd
import numpy as np
import matplotlib.pyplot as plt
import numpy as np
import scipy.stats as st

In [ ]:
fk_sales=pd.read_csv('/content/drive/MyDrive/flipkart_com-ecommerce_sample.csv')
fk_sales

uniq_id            crawl_timestamp  \
0      c2d766ca982eca8304150849735ffef9  2016-03-25 22:59:23 +0000   
1      7f7036a6d550aaa89d34c77bd39a5e48  2016-03-25 22:59:23 +0000   
2      f449ec65dcbc041b6ae5e6a32717d01b  2016-03-25 22:59:23 +0000   
3      0973b37acd0c664e3de26e97e5571454  2016-03-25 22:59:23 +0000   
4      bc940ea42ee6bef5ac7cea3fb5cfbee7  2016-03-25 22:59:23 +0000   
...                                 ...                        ...   
19997  93e9d343837400ce0d7980874ece471c  2015-12-01 10:15:43 +0000   
19998  669e79b8fa5d9ae020841c0c97d5e935  2015-12-01 10:15:43 +0000   
19999  cb4fa87a874f715fff567f7b7b3be79c  2015-12-01 10:15:43 +0000   
20000                               NaN                        NaN   
20001                               NaN                        NaN   

                                             product_url  \
0      http://www.flipkart.com/alisha-solid-women-s-c...   
1      http://www.flipkart.com/fabhomedecor-fabric-do...   
2      http://www.flipkart.com/aw-bellies/p/itmeh4grg...   
3      http://www.flipkart.com/alisha-solid-women-s-c...   
4      http://www.flipkart.com/sicons-all-purpose-arn...   
...                                                  ...   
19997  http://www.flipkart.com/elite-collection-mediu...   
19998  http://www.flipkart.com/elite-collection-mediu...   
19999  http://www.flipkart.com/elite-collection-mediu...   
20000                                                NaN   
20001                                                NaN   

                                  product_name  \
0          Alisha Solid Women's Cycling Shorts   
1          FabHomeDecor Fabric Double Sofa Bed   
2                                   AW Bellies   
3          Alisha Solid Women's Cycling Shorts   
4        Sicons All Purpose Arnica Dog Shampoo   
...                                        ...   
19997  Elite Collection Medium Acrylic Sticker   
19998  Elite Collection Medium Acrylic Sticker   
19999  Elite Collection Medium Acrylic Sticker   
20000                                      NaN   
20001                                      NaN   

                                   product_category_tree               pid  \
0      ["Clothing >> Women's Clothing >> Lingerie, Sl...  SRTEH2FF9KEDEFGF   
1      ["Furniture >> Living Room Furniture >> Sofa B...  SBEEH3QGU7MFYJFY   
2      ["Footwear >> Women's Footwear >> Ballerinas >...  SHOEH4GRSUBJGZXE   
3      ["Clothing >> Women's Clothing >> Lingerie, Sl...  SRTEH2F6HUZMQ6SJ   
4      ["Pet Supplies >> Grooming >> Skin & Coat Care...  PSOEH3ZYDMSYARJ5   
...                                                  ...               ...   
19997  ["Baby Care >> Baby & Kids Gifts >> Stickers >...  STIE7VAYDKQZEBSD   
19998  ["Baby Care >> Baby & Kids Gifts >> Stickers >...  STIE8YSVEPPCZ42Y   
19999  ["Baby Care >> Baby & Kids Gifts >> Stickers >...  STIE88KN9ZDSGZKY   
20000                                                NaN               NaN   
20001                                                NaN               NaN   

       retail_price  discounted_price  \
0             999.0             379.0   
1           32157.0           22646.0   
2             999.0             499.0   
3             699.0             267.0   
4             220.0             210.0   
...             ...               ...   
19997        1299.0             999.0   
19998        1499.0            1199.0   
19999        1499.0             999.0   
20000           NaN               NaN   
20001           NaN               NaN   

                                                   image  \
0      ["http://img5a.flixcart.com/image/short/u/4/a/...   
1      ["http://img6a.flixcart.com/image/sofa-bed/j/f...   
2      ["http://img5a.flixcart.com/image/shoe/7/z/z/r...   
3      ["http://img5a.flixcart.com/image/short/6/2/h/...   
4      ["http://img5a.flixcart.com/image/pet-shampoo/...   
...                                                  ...   
19997  ["http://img

In [ ]:
fk_sales.columns

Index(['uniq_id', 'crawl_timestamp', 'product_url', 'product_name',
       'product_category_tree', 'pid', 'retail_price', 'discounted_price',
       'image', 'is_FK_Advantage_product', 'description', 'product_rating',
       'overall_rating', 'brand', 'product_specifications'],
      dtype='object')

In [ ]:
df=pd.DataFrame({'product_name':fk_sales['product_name'],'retail_price':fk_sales['retail_price'],'discounted_price':fk_sales['discounted_price'],'brand':fk_sales['brand'],})
df['product_name']=fk_sales['product_name']
df

product_name  retail_price  \
0          Alisha Solid Women's Cycling Shorts         999.0   
1          FabHomeDecor Fabric Double Sofa Bed       32157.0   
2                                   AW Bellies         999.0   
3          Alisha Solid Women's Cycling Shorts         699.0   
4        Sicons All Purpose Arnica Dog Shampoo         220.0   
...                                        ...           ...   
19997  Elite Collection Medium Acrylic Sticker        1299.0   
19998  Elite Collection Medium Acrylic Sticker        1499.0   
19999  Elite Collection Medium Acrylic Sticker        1499.0   
20000                                      NaN           NaN   
20001                                      NaN           NaN   

       discounted_price             brand  
0                 379.0            Alisha  
1               22646.0      FabHomeDecor  
2                 499.0                AW  
3                 267.0            Alisha  
4                 210.0            Sicons  
...                 ...               ...  
19997             999.0  Elite Collection  
19998            1199.0  Elite Collection  
19999             999.0  Elite Collection  
20000               NaN               NaN  
20001               NaN               NaN  

[20002 rows x 4 columns]

In [ ]:
fk_sales.describe()

retail_price  discounted_price
count   19922.000000      19922.000000
mean     2979.206104       1973.401767
std      9009.639341       7333.586040
min        35.000000         35.000000
25%       666.000000        350.000000
50%      1040.000000        550.000000
75%      1999.000000        999.000000
max    571230.000000     571230.000000

In [ ]:
fk_sales[('discounted_price')].mean()

1973.4017668908743

In [ ]:
sample=df.loc[df['product_name']=='Elite Collection Medium Acrylic Sticker']
samplec=sample.sample(1000,replace=True)
samplec

product_name  retail_price  \
19847  Elite Collection Medium Acrylic Sticker        1299.0   
19865  Elite Collection Medium Acrylic Sticker        1299.0   
19999  Elite Collection Medium Acrylic Sticker        1499.0   
19806  Elite Collection Medium Acrylic Sticker        1299.0   
19806  Elite Collection Medium Acrylic Sticker        1299.0   
...                                        ...           ...   
19935  Elite Collection Medium Acrylic Sticker        1299.0   
19998  Elite Collection Medium Acrylic Sticker        1499.0   
19998  Elite Collection Medium Acrylic Sticker        1499.0   
19944  Elite Collection Medium Acrylic Sticker        1299.0   
19894  Elite Collection Medium Acrylic Sticker        1599.0   

       discounted_price             brand  
19847             999.0  Elite Collection  
19865             999.0  Elite Collection  
19999             999.0  Elite Collection  
19806             999.0  Elite Collection  
19806             999.0  Elite Collection  
...                 ...               ...  
19935             999.0  Elite Collection  
19998            1199.0  Elite Collection  
19998            1199.0  Elite Collection  
19944             999.0  Elite Collection  
19894            1199.0  Elite Collection  

[1000 rows x 4 columns]

# Hypothesis testing

 H0:Discounted price makes an averge  of 2979.206104



 
 H1: Discounted price does not makes an averge  of 2979.206104

In [ ]:
samplec.discounted_price.mean()

1008.65

### z test

In [ ]:
#standard deviation of the population
sd=np.std(samplec['discounted_price'])
sd

139.99777676806156

In [ ]:
samplec.std()

<ipython-input-22-4d62aafc4dea>:1: FutureWarning: Dropping of nuisance columns in DataFrame reductions (with 'numeric_only=None') is deprecated; in a future version this will raise TypeError.  Select only valid columns before calling the reduction.
  samplec.std()


retail_price        176.887080
discounted_price    140.067828
dtype: float64

In [ ]:
xbar=samplec.discounted_price.mean()
mu=2979.20
n=1000 #sample size
alpha=0.05

In [ ]:
xbar

1008.65

In [ ]:
zvalue=(xbar-mu)/(sd/np.sqrt(n))
zvalue

-445.1089429490431

In [ ]:
z_critical = st.norm.ppf(alpha/2)  #One tail Test
z_critical

-1.9599639845400545

In [ ]:
if (zvalue<z_critical and abs(zvalue)> abs(z_critical)):
  print("Reject Null Hypotheisis")
else:
  print("Fail to reject Null Hypothesis")

Reject Null Hypotheisis


## using p value

In [ ]:
p_val = (1 - st.norm.cdf(abs(zvalue))) * 2
p_val

0.0

In [ ]:
#comparing with alpha value
if p_val<alpha:
  print("Reject Null Hypotheisis")
else:
  print("Fail to reject Null Hypothesis")

Reject Null Hypotheisis


## Hence discounted price on Average doesn't make 2979.20, i.e, we reject the Null Hypothesis

### T test

In [ ]:
tsamplec=sample.sample(28,replace=True)
tsamplec

product_name  retail_price  \
19863  Elite Collection Medium Acrylic Sticker        1999.0   
19789  Elite Collection Medium Acrylic Sticker        1299.0   
19944  Elite Collection Medium Acrylic Sticker        1299.0   
19964  Elite Collection Medium Acrylic Sticker        1299.0   
19863  Elite Collection Medium Acrylic Sticker        1999.0   
19805  Elite Collection Medium Acrylic Sticker        1299.0   
19917  Elite Collection Medium Acrylic Sticker        1299.0   
19998  Elite Collection Medium Acrylic Sticker        1499.0   
19945  Elite Collection Medium Acrylic Sticker        1499.0   
19922  Elite Collection Medium Acrylic Sticker        1299.0   
19894  Elite Collection Medium Acrylic Sticker        1599.0   
19917  Elite Collection Medium Acrylic Sticker        1299.0   
19880  Elite Collection Medium Acrylic Sticker        1299.0   
19839  Elite Collection Medium Acrylic Sticker        1299.0   
19984  Elite Collection Medium Acrylic Sticker        1299.0   
19964  Elite Collection Medium Acrylic Sticker        1299.0   
19937  Elite Collection Medium Acrylic Sticker         999.0   
19999  Elite Collection Medium Acrylic Sticker        1499.0   
19997  Elite Collection Medium Acrylic Sticker        1299.0   
19839  Elite Collection Medium Acrylic Sticker        1299.0   
19993  Elite Collection Medium Acrylic Sticker        1299.0   
19823  Elite Collection Medium Acrylic Sticker        1299.0   
19803  Elite Collection Medium Acrylic Sticker        1299.0   
19823  Elite Collection Medium Acrylic Sticker        1299.0   
19944  Elite Collection Medium Acrylic Sticker        1299.0   
19826  Elite Collection Medium Acrylic Sticker         999.0   
19922  Elite Collection Medium Acrylic Sticker        1299.0   
19864  Elite Collection Medium Acrylic Sticker        1299.0   

       discounted_price             brand  
19863            1399.0  Elite Collection  
19789             999.0  Elite Collection  
19944             999.0  Elite Collection  
19964             999.0  Elite Collection  
19863            1399.0  Elite Collection  
19805             999.0  Elite Collection  
19917             999.0  Elite Collection  
19998            1199.0  Elite Collection  
19945            1199.0  Elite Collection  
19922             999.0  Elite Collection  
19894            1199.0  Elite Collection  
19917             999.0  Elite Collection  
19880             999.0  Elite Collection  
19839             999.0  Elite Collection  
19984             999.0  Elite Collection  
19964             999.0  Elite Collection  
19937             699.0  Elite Collection  
19999             999.0  Elite Collection  
19997             999.0  Elite Collection  
19839             999.0  Elite Collection  
19993             999.0  Elite Collection  
19823             999.0  Elite Collection  
19803             999.0  Elite Collection  
19823             999.0  Elite Collection  
19944             999.0  Elite Collection  
19826             599.0  Elite Collection  
19922             999.0  Elite Collection  
19864             999.0  Elite Collection

In [ ]:
t_sd=np.std(tsamplec['discounted_price'])
t_sd

154.97695681247953

In [ ]:
xbar=tsamplec.discounted_price.mean()
print(xbar)
mu=2979.20
n=28 #sample size
alpha=0.05

1024.0


In [ ]:
n = 28
degrees_of_freedom = n-1
t = (xbar - mu)/(t_sd / np.sqrt(n))
t

-66.75796285834583

In [ ]:
t_critical = st.t.ppf(alpha, degrees_of_freedom)
t_critical

-1.703288445722127

In [ ]:
if (t > t_critical):
    print("Null hypothesis cannot be rejected")
else:
    print("Reject null hypothesis")

Reject null hypothesis


### Both Z test and T test suggest the same result, The Null Hypothesis can be rejected.